cmd/desync/inspectchunks.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"github.com/folbricht/desync"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+)
+
+type inspectChunksOptions struct {
+	cmdStoreOptions
+	store string
+}
+
+func newinspectChunksCommand(ctx context.Context) *cobra.Command {
+	var opt inspectChunksOptions
+
+	cmd := &cobra.Command{
+		Use:   "inspect-chunks <index> [<output>]",
+		Short: "Inspect chunks from an index and an optional local store",
+		Long: `Prints a detailed JSON with information about chunks stored in an index file.
+By using the '--store' option to provide a local store, the generated JSON will include, if
+available, the chunks compressed size info from that particular store.`,
+		Example: `  desync inspect-chunks file.caibx
+desync inspect-chunks --store /mnt/store file.caibx inspect_result.json`,
+		Args: cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runInspectChunks(ctx, opt, args)
+		},
+		SilenceUsage: true,
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&opt.store, "store", "s", "", "local source store")
+	addStoreOptions(&opt.cmdStoreOptions, flags)
+	return cmd
+}
+
+func runInspectChunks(ctx context.Context, opt inspectChunksOptions, args []string) error {
+	if err := opt.cmdStoreOptions.validate(); err != nil {
+		return err
+	}
+
+	var (
+		outFile io.Writer
+		err     error
+	)
+	if len(args) == 2 {
+		outFileName := args[1]
+		outFile, err = os.Create(outFileName)
+		if err != nil {
+			return err
+		}
+	} else {
+		outFile = stdout
+	}
+
+	// Read the input
+	c, err := readCaibxFile(args[0], opt.cmdStoreOptions)
+	if err != nil {
+		return err
+	}
+
+	var chunksInfo []desync.ChunkAdditionalInfo
+
+	var s desync.LocalStore
+	if opt.store != "" {
+		options, err := cfg.GetStoreOptionsFor(opt.store)
+		if err != nil {
+			return err
+		}
+
+		// Use compressed chunks, if available, to record their sizes
+		options.Uncompressed = false
```

Comment:
This doesn't look quite right and is confusing for the user. We should allow the user to specify if the store is compressed or not and not override configuration. Wouldn't it be better to simply skip `size, _ = s.GetChunkSize(chunk.ID)` in the loop below when `options.Uncompressed` is true?

Reply:
Yeap, now I don't override that option and I added `!s.Opt.Uncompressed` to the `if`.